client/src/app/app.module.ts

```diff
@@ -0,0 +1,48 @@
+import { NgModule } from '@angular/core';
+import { BrowserModule } from '@angular/platform-browser';
+import { FormsModule, ReactiveFormsModule } from '@angular/forms'; 
+import { HttpClientModule } from '@angular/common/http';
+import { HttpClientInMemoryWebApiModule } from 'angular-in-memory-web-api';
+import { InMemoryDataService } from './in-memory-data.service';
+
+import { AppRoutingModule } from './app-routing.module';
+import { AppComponent } from './app.component';
+import { UserlistComponent } from './userlist/userlist.component';
+import { UserDetailComponent } from './user-detail/user-detail.component';
+import { MessagesComponent } from './messages/messages.component';
+import { DashboardComponent } from './dashboard/dashboard.component';
+import { EditviewComponent } from './editview/editview.component';
+import { AutofocusDirective } from './autofocus.directive';
+import { AddviewComponent } from './addview/addview.component';
+import { ChangedeleteviewComponent } from './changedeleteview/changedeleteview.component';
+
+@NgModule({
+  declarations: [
+    AppComponent,
+    UserlistComponent,
+    UserDetailComponent,
+    MessagesComponent,
+    DashboardComponent,
+    EditviewComponent,
+    AutofocusDirective,
```

Comment:
Directives or other shared ui components should be exported from their own module